packages/react-widgets/src/widgets/FormulaWidget.js

```diff
@@ -30,17 +29,19 @@ function FormulaWidget(props) {
     onError,
     wrapperProps
   } = props;
+  const isSourceReady = useSelector(
```

Comment:
We repeat this code in all widgets. Could we have a redux selector?

Reply:
What do you think @VictorVelarde? I am in favor of adding a selector in redux, but I saw it isn't common in c4r, then I didn't do it.

Reply:
Just below you have a selector XD `selectSourceById`

Reply:
`selectViewportFeaturesReadyBySourceId` is a good name? Maybe too long

Reply:
Verbose names is a good practice, another more descriptive could be `selectIsViewportFeaturesReadyForSource`

Reply:
Added.

Reply:
I like Javi's naming proposal